TestApp/src/main/java/com/tekchand/testapp/ui/main/LogIn.java

```diff
@@ -0,0 +1,99 @@
+package com.tekchand.testapp.ui.main;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.google.android.material.textfield.TextInputLayout;
+import com.tekchand.testapp.MainActivity;
+import com.tekchand.testapp.R;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+public class LogIn extends AppCompatActivity {
```

Comment:
Since this is an activity, Append "Activity" to the class name.

Reply:
done. "LogInActivity"